specification/hugr.md

```diff
@@ -998,19 +1002,20 @@ at runtime. In many cases this is desirable.
 To strike a balance then, every extension provides declarative structs containing
 named **TypeDef**s and **OpDef**s---see [Declarative Format](#declarative-format).
 These are (potentially polymorphic) definitions of types and operations, respectively---polymorphism arises because both may
-declare any number of TypeParams (as per [Type System](#type-system)). To use a TypeDef as a type,
+declare any number of TypeParams, as per [Polymorphism](#polymorphism). To use a TypeDef as a type,
 it must be instantiated with TypeArgs appropriate for its TypeParams, and similarly
 to use an OpDef as a node operation: each `OpaqueOp` node stores a static-constant list of TypeArgs.
 
 For TypeDef's, any set of TypeArgs conforming to its TypeParams, produces a valid type.
 However, for OpDef's, greater flexibility is allowed: each OpDef *either*
 
-1. Provides a `Function` type, as per [Type System](#type-system), which may declare TypeParams; *or*
+1. Provides a polymorphic type scheme, as per [Type System](#type-system), which may declare TypeParams;
+   values (TypeArgs) provided for those params will be substituted in. *Or*
 2. The extension may self-register binary code (e.g. a Rust trait) providing a function
-   `compute_signature` that fallibly computes a `Function` type given some type arguments.
+   `compute_signature` that fallibly computes a (perhaps-polymorphic) type scheme given some initial type arguments.
    The operation declares the arguments required by the `compute_signature` function as a list
-   of TypeParams; if this successfully returns a `Function` type, then that may then require
-   additional TypeParams.
+   of TypeParams; if this successfully returns a typ scheme, that may have additional TypeParams
```

Comment:
```suggestion
   of TypeParams; if this successfully returns a type scheme, that may have additional TypeParams
```

Reply:
Ha (thank you!) - maybe I meant a tip scheme, for dividing gratuities between team members